server/odc-service/src/main/java/com/oceanbase/odc/service/session/OdcStatementCallBack.java

```diff
@@ -158,50 +174,75 @@ public List<JdbcGeneralResult> doInStatement(Statement statement) throws SQLExce
             if (this.autoCommit ^ currentAutoCommit) {
                 statement.getConnection().setAutoCommit(this.autoCommit);
             }
+            List<JdbcGeneralResult> returnVal = new LinkedList<>();
+            Future<Void> handle = null;
             for (SqlTuple sqlTuple : this.sqls) {
+                if (handle != null) {
+                    try {
+                        handle.get();
+                    } catch (Exception e) {
+                        // eat exception
+                    }
+                }
+                if (context != null) {
+                    context.onQueryStart(sqlTuple.getExecutedSql());
+                }
                 try {
                     applyConnectionSettings(statement);
                 } catch (Exception e) {
                     log.warn("Init driver statistic collect failed, reason={}", e.getMessage());
                 }
                 List<JdbcGeneralResult> executeResults;
                 if (returnVal.stream().noneMatch(r -> r.getStatus() == SqlExecuteStatus.FAILED) || !stopWhenError) {
-                    try {
-                        executeResults = doExecuteSql(statement, sqlTuple);
-                    } catch (Exception exception) {
-                        executeResults = Collections.singletonList(JdbcGeneralResult.failedResult(sqlTuple, exception));
+                    CountDownLatch latch = new CountDownLatch(1);
+                    if (context != null) {
+                        handle = executor.submit(() -> {
+                            try {
+                                if (!latch.await(1100, TimeUnit.MILLISECONDS)) {
+                                    context.onQueryExecuting();
+                                }
+                            } catch (InterruptedException e) {
+                                log.warn("Failed to call back.", e);
+                            }
+                            return null;
+                        });
                     }
+                    executeResults = doExecuteSql(statement, sqlTuple, latch);
                 } else {
                     executeResults = Collections.singletonList(JdbcGeneralResult.canceledResult(sqlTuple));
                 }
                 returnVal.addAll(executeResults);
+                if (context != null) {
+                    context.onQueryFinish(executeResults);
+                }
             }
             Optional<JdbcGeneralResult> failed = returnVal
                     .stream().filter(r -> r.getStatus() == SqlExecuteStatus.FAILED).findFirst();
             if (failed.isPresent()) {
-                throw failed.get().getThrown();
-            }
-
-        } catch (Exception e) {
-            try {
-                ConnectionSessionUtil.logSocketInfo(statement.getConnection(), "console error");
-            } catch (Exception exception) {
-                log.warn("Failed to execute abnormal replenishment logic", exception);
+                try {
```

Comment:
we's better not do this, cause that we can not guarantee that there is no exception thrown in try-catch

Reply:
ok, restored the original logic